### DISCUSSION TITLE
Welcome to XppTools Discussions!

### DISCUSSION THREAD

**Discussion by TrudAX** [TrudAX/XppTools#4] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by supersaad01** [comment]
Hi! great post thank you so much!
I have a question.
I installed the models in my dev environemtn.
Now I must deploy into UAT and PROD cloud environements. How can I do that?

**Reply by TrudAX** [reply]
Check-in into the version control, then create a package and install it on UAT..
https://docs.microsoft.com/en-us/dynamics365/fin-ops-core/dev-itpro/deployment/create-apply-deployable-package

**Reply by supersaad01** [reply]
To do a checkin I must create a  VS solution and add all model objects and add them to version control?

**Reply by TrudAX** [reply]
Yes. or you can just add files manually from Source Control Explorer
https://docs.microsoft.com/en-us/azure/devops/repos/tfvc/add-files-server?view=azure-devops#manually-add-files-to-version-control

**Reply by supersaad01** [reply]
thank you so much!

**Comment by medhaouari77** [comment]
Thank you so much for this really helpful.
I'm using DEVSQLExecute and i'm trying to add a menu item button in Form information(FormControlPersonalization) under query string under the Query statment field 
![image](https://user-images.githubusercontent.com/85366972/222464155-9dbfd75f-5b65-4a82-a490-018204cc6bed.png)
when clicked the button will open the DEVSQLQueryExecute and pass Query statment to QueryText is it possible ?

**Reply by TrudAX** [reply]
Good idea. The problem is X++ query syntax(that you see in the Query statement field) is different from SQL syntax. If you find a way to convert it, that would be great

**Reply by medhaouari77** [reply]
ah okey i've checked out, thank you for clarifying that.
So i will just try to get the DataSource name and make the SQLText in the init() and change it to"select * from (DataSource name)"
any idea how can i get the DataSource field value?

**Comment by bradlyjames** [comment]
HI there! So far these have been some good tools. One question, after building the model and syncing the model, I still don't see the security roles you created. Is there a step I'm missing?

**Reply by TrudAX** [reply]
Have you done the full sync? Roles should be created as a last step of sync

**Reply by bradlyjames** [reply]
Ahhh. That's what I missed.  I did the 'invoke' sync command on the model but not a full sync.  Doing that now.  Thanks!

**Reply by bradlyjames** [reply]
Confirmed the full DB sync worked. Thanks!

**Comment by aamirhanif1** [comment]
Hi, Thanks for making our life easier. I just wanted to point out one thing.
[DEVDMFTmpEntityList.xml](https://github.com/TrudAX/XppTools/blob/master/DEVTools/DEVDMFTools/AxTable/DEVDMFTmpEntityList.xml)
is missing many fields.

**Reply by TrudAX** [reply]
Thanks. What fields do this think are missed?

**Comment by ripannandy** [comment]
Service bus integration section really helped me a lot. Thank you so much for this post.

Do you have any documentation like Service bus integration for Azure Event Grid?

**Reply by TrudAX** [reply]
Hi Ripan. Thanks. I have not done Azure Event Grid yet If you plan to do it - it can be done by extending the DEVIntegMessagesLoadBaseType  class
